ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```diff
@@ -0,0 +1,559 @@
+        return True
+
+    @cached_property
+    def ipv6_unicast_routing(self):
+        if get(self._hostvars, 'underlay_rfc5549') is True or get(self._switch_data, 'underlay_ipv6') is True:
+            return True
+        return None
+
+    @cached_property
+    def ip_routing_ipv6_interfaces(self):
+        if get(self._hostvars, 'underlay_rfc5549') is True:
+            return True
+        return None
+
+    @cached_property
+    def router_multicast(self):
+        if (
+                get(self._switch_data, 'underlay_multicast') is True and
+                get(self._switch_data, 'underlay_router') is True
+        ):
+            router_multicast = {'ipv4': {'routing': True}}
+            if get(self._switch_data, 'evpn_multicast') is True:
+                router_multicast['ipv4'].update({'software_forwarding': 'sfe'})
+            return router_multicast
+        return None
+
+    @cached_property
+    def hardware_counters(self):
+        hardware_counter = get(self._hostvars, 'hardware_counters')
+        if hardware_counter is not None:
+            features = []
+            if get(hardware_counter, 'features') is not None:
+                features = [{feature: direction} for feature, direction in get(hardware_counter, 'features').items()]
+            return {"features": features}
+        return None
+
+    @cached_property
+    def hardware(self):
+        platform_speed_groups = get(self._hostvars, 'platform_speed_groups')
+        if platform_speed_groups is not None:
+            tmp_speed_groups = {}
+            switch_platform = get(self._switch_data, 'platform')
+            if switch_platform is not None:
+                for platform_item in platform_speed_groups:
+                    if "platform" in platform_item and "speeds" in platform_item:
+                        platform = platform_item
+                    else:
+                        platform = {"platform": platform_item, "speeds": platform_speed_groups[platform_item]}
+                    if platform['platform'] == switch_platform:
+                        speeds = self._convert_dicts(get(platform, 'speeds'), 'speed', 'speed_groups')
+                        for speed in self._natural_sort(speeds, 'speed'):
+                            for speed_group in speed['speed_groups']:
+                                tmp_speed_groups.update({speed_group: speed['speed']})
+            if tmp_speed_groups:
+                hardware = {'speed_groups': {}}
+                for speed_group in self._natural_sort(tmp_speed_groups):
+                    hardware['speed_groups'].update({speed_group: {'serdes': tmp_speed_groups[speed_group]}})
+                return hardware
+        return None
+
+    @cached_property
+    def daemon_terminattr(self):
+        cvp_instance_ip = get(self._hostvars, 'cvp_instance_ip')
+        cvp_instance_ips = get(self._hostvars, 'cvp_instance_ips')
+        if cvp_instance_ip is not None or cvp_instance_ips is not None:
```

Comment:
Lets return None to decrease indentation of the method as in other methods

```suggestion
        if cvp_instance_ip is None and cvp_instance_ips is None:
          return None
        cvp_on_prem = {"ips": []}
        ...
```